### DISCUSSION TITLE
Broken dependnecy to pytorch_lightning?

### DISCUSSION THREAD

**Discussion by dgoryeo** [NVIDIA-NeMo/NeMo#6577] [Q&A]
I cannot find a workaround with the error below while running in Google Colab. I have installed pytorch-lightning with command  `!pip install -qq pytorch-lightning==1.5.4` and have tried many different versions too. But still import fails. What is going wrong? 

```

---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
[<ipython-input-14-99c1ac25fe61>](https://localhost:8080/#) in <cell line: 11>()
      9 import librosa
     10 import soundfile
---> 11 from nemo.collections.asr.models.msdd_models import NeuralDiarizer
     12 from deepmultilingualpunctuation import PunctuationModel
     13 import re

5 frames
[/usr/local/lib/python3.10/dist-packages/nemo/core/classes/exportable.py](https://localhost:8080/#) in <module>
     16 
     17 import torch
---> 18 from pytorch_lightning.core.module import _jit_is_scripting
     19 
     20 from nemo.core.classes import typecheck

ModuleNotFoundError: No module named 'pytorch_lightning.core.module'

---------------------------------------------------------------------------

```

**Comment by titu1994** [comment]
You have to use the pytorch lightning version that Nemo is compatible with. The latest Nemo branch supports lightning 1.9 and above (but not 2.0 yet)

**Comment by dgoryeo** [comment]
@titu1994 , I have tried: 1.5.5 , 1.6.5, 1.7.7, and 1.9.3. They all fail.  My torch is 1.13.1.

**Reply by titu1994** [reply]
Which Nemo version are you trying, and you should not manually install or update lightning. Simply install Nemo toolkit all to install the correct dependencies

**Comment by dgoryeo** [comment]
@titu1994 , I'm using version 1.17.0  `!pip install -qq nemo_toolkit[asr]==1.17.0`

And here is the notebook that I am trying to use:  [Whisper_Transcription_+_NeMo_Diarization.ipynb](https://github.com/MahmoudAshraf97/whisper-diarization/blob/main/Whisper_Transcription_%2B_NeMo_Diarization.ipynb)

**Reply by titu1994** [reply]
I ran your notebook on colab and it seems to install properly (import works for NeMo modules). Try pip uninstall pytorch_lightning and then reinstall NeMo locally

**Reply by dgoryeo** [reply]
Thanks @titu1994 , I did try uninstall but the error persists. The error is from a version/requirement mismatch between pyannote and nemo. I isolated the problem by removing the package that has dependency to pyannote. I have still not solved the problem but at least I can now proceed.

**Reply by titu1994** [reply]
Interesting, could you clarify what dependency of pyanotate is causing the issue ? Does it depend on lightning too ?
